src/docfx/restore/PackageResolver.cs

```diff
@@ -151,8 +152,10 @@ private void DownloadGitRepositoryCore(string cwd, string url, string committish
             {
                 GitUtility.Fetch(_config, cwd, url, $"+{committish}:{committish}", $"{fetchOption} {depthOneOption}");
             }
-            catch (InvalidOperationException)
+            catch (InvalidOperationException invalidOperationException)
             {
+                ThrowRestoreDependentRepositoryFailureError(url, committish, invalidOperationException);
```

Comment:
move this check to line 165 as there is a retry logic in line 162

Reply:
Why do we want to retry if there's SSO/permission exception

Reply:
This is a general retry regardless of SSO. The `committish` can be a short git hash that can you cannot retrieve with `+bla:bla` and it needs to fallback to using `+refs/heads/*:refs/heads`. Line 157 tries to fetch as less as possible.